### DISCUSSION TITLE
Authentication

### DISCUSSION THREAD

**Discussion by melwinek** [music-assistant/support#1544] [Q&A] [answered]
I'm using Music Assistant Server as a Docker container. Currently, anyone on the local network can access the website and control playback. Is it possible to force login to the website?

**Comment by OzGav** [comment]
No. We recommend using HAOS for these security reasons

**Comment by marcelveldt** [comment, marked as answer]
When running the docker container manually yourself, it is your own responsibility to secure the webinterface, e.g. by putting a reverse proxy in front. We offer some settings to only start the webserver on a private/internal docker network.

By far the most people run MA within Home Assistant OS where the MA webinterfacce is hosted securely behind the ingress.
Implementing an own auth system for Music Assistant itself will be auth of scope for now unless maybe we can find a quick way to leverage HA auth on the MA webinterface.

**Reply by antialias** [reply]
@marcelveldt You can leverage HA for auth on the MA webinterface using their OAuth 2 flow:
https://developers.home-assistant.io/docs/auth_api/

I put MA behind a reverse proxy as you suggested and it was working well up until I got the latest version of HA and now it doesn't like my redirect URI. No helpful information in the logs, and I'm tired of maintaining this just for my instance. Will probably stop using MA until y'all can figure this this out as it's currently a bit of a deal-breaker.

Requiring HA OS is pretty lame.

**Reply by marcelveldt** [reply]
Your comment is lame imo and not a good motivator to actually pick this up from the pile of work.

**Reply by antialias** [reply]
@marcelveldt well your reply is lame too and furthermore I'm not gonna invite you to my birthday party

**Reply by antialias** [reply]
with an attitude like that for receiving feedback, no wonder MA is an insecure pile of $*@#

**Reply by marcelveldt** [reply]
Bye bye you have just been blocked entirely - what an attitude as gratitude for a product you can use for free.

Some people truly have no idea how an open source community works and think they can just claim and complain.

Remember this has costed me 5 years of hard work and time away from family, trying to listen to most feedback and incorporate the features that are appreciated by MOST folks - some features will not be implemented by me because my priority is needed elsewhere but ofc others can chime in and PR it.

I suggest folks to give constructive feedback and calling stuff lame or a piece of shit is not going to get you anywhere.

**Comment by XDROLLOXD** [comment]
Hi, 

there thanks for the nice project,
i am using HomeAssistant for three apartments.
Is there a way to set up some permissions?
like user_1 is able to controll device 1 - 4 
and user_2 is only able to controll device 3 and 4. 
It is also possible for each user to use their own streaming services?
Like: User_1 = Spotify_account_1 and user_2 = Spotify_account_2
I'm thinking of a small user administration.

Thank you and have a nice day :)

**Reply by marcelveldt** [reply]
No that is not possible and is most likely not something we will add in the near future

**Reply by OzGav** [reply]
Best you could do for the first part is setup user restricted views in HA that only show certain players but obviously you won’t get the full MA UI

**Reply by marcelveldt** [reply]
Yeah, that is also how I do this myself. The kids just have their own HA dashboard with a few cards for music playback.

**Comment by SteveDinn** [comment]
I was looking for authentication as well. and found this issue. Before I go through the trouble of setting up Authelia or something, is there any way that a stand-alone MusicAssistant container can be integrated into HA like the proper add-on? I run HA as a container as well, not the full HA operating system, so I fear I may be out of luck, but thought I'd ask the people who would know.

**Reply by OzGav** [reply]
Addons can only be used with HAOS

**Comment by NeviZz** [comment]
Hi,
I’ve just started using Music Assistant (it’s brilliant!), but I’m facing the same issue as the colleagues above. Basically, anyone who knows the Home Assistant link and the port on which MA runs can control it.
Do you plan to implement some form of authentication and permissions? I think that would be a good way to secure the page. :)

**Reply by marcelveldt** [reply]
Yes, that is what is been added and is in the current 2.7 beta which goes to stable this Wednesday

**Reply by NeviZz** [reply]
NICE! Thank you for that fast answer! :D

**Comment by OzGav** [comment]
Doing a sweep and closing old outdated posts
